src/compiler/crystal/loader/unix.cr

```diff
@@ -30,7 +30,7 @@ class Crystal::Loader
   def self.parse(args : Array(String), *, search_paths : Array(String) = default_search_paths) : self
     libnames = [] of String
     file_paths = [] of String
-    OptionParser.parse(args) do |parser|
+    OptionParser.parse(args.dup) do |parser|
```

Comment:
I don't understand this change. Why is it needed?

Reply:
Option parser removes items from the array as it processes them. If we let that happen, we wouldn't get a meaningful error message.